Development/nmos/control_protocol_state.h

```diff
@@ -0,0 +1,99 @@
+#ifndef NMOS_CONTROL_PROTOCOL_STATE_H
+#define NMOS_CONTROL_PROTOCOL_STATE_H
+
+#include <map>
+#include "cpprest/json_utils.h"
+#include "nmos/control_protocol_handlers.h"
+#include "nmos/control_protocol_typedefs.h"
+#include "nmos/mutex.h"
+
+namespace slog { class base_gate; }
+
+namespace nmos
+{
+    namespace experimental
+    {
+        struct control_class // NcClassDescriptor
```

Comment:
To avoid confusion between class definitions and instances 

```suggestion
        struct control_class_descriptor // NcClassDescriptor
```

Reply:
done